Refactor the code for AWS authentication

lerks opened the pull request:
Stop using the WSGI environment and the headers dictionary to communicate with the authentication middleware. Instead, store it as a service-level attribute, so that the application can access the instance and interact with a proper API.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cms-dev/cms/623)
<!-- Reviewable:end -->

stefano-maggiolo commented:
Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[cms/io/web_service.py, line 71 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKo8sJJfrRJbqxwY78F:-KKo8sJL4SiXZUCe6eE4:278645343) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/io/web_service.py#L71)):*
> ```Python
>                 self.wsgi_app, {"/rpc": RPCMiddleware(self, rpc_auth)})
> 
>         # It needs to be applied before the ProxyFix for the remote
> ```

What's "it"?

---

*[cms/io/web_service.py, line 72 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKo8xDaJg6Hc-u5TF1p:-KKo8xDaJg6Hc-u5TF1q:-1654800733) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/io/web_service.py#L72)):*
> ```Python
> 
>         # It needs to be applied before the ProxyFix for the remote
>         # address to be properly recognized.
> ```

Make the comment more useful, explaining why, or what "proper" means?

---

*[cms/server/admin/authentication.py, line 74 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKoApqUT81cPK-3zHYF:-KKoApqUT81cPK-3zHYG:677121513) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/server/admin/authentication.py#L74)):*
> ```Python
> 
>         It's the value that has been read from the cookie and (if
>         modified by the rest of the application )will be written back.
> ```

Typo

---

*[cms/server/admin/authentication.py, line 147 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKoBZ1aUnjIHNOgTjMp:-KKoBZ1aUnjIHNOgTjMq:1473337632) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/server/admin/authentication.py#L147)):*
> ```Python
> 
>     def _verify_cookie(self):
>         """Perform sanity checks on the stored cookie.
> ```

"""Check that the cookie is valid, and if not clear it"""

---

*[cms/server/admin/authentication.py, line 150 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKoBpWxJKbtLlVLqWaf:-KKoBpWxJKbtLlVLqWag:-944040243) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/server/admin/authentication.py#L150)):*
> ```Python
> 
>         """
>         if len(self._cookie) == 0:
> ```

What's this for?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/623)*
<!-- Sent from Reviewable.io -->

lerks commented:
Review status: 6 of 8 files reviewed at latest revision, 6 unresolved discussions.

---

*[cms/io/web_service.py, line 71 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKo8sJJfrRJbqxwY78F:-KKpYXVK9eOhcMooqxBP:1758359711) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/io/web_service.py#L71)):*
<details><summary><i>Previously, stefano-maggiolo (Stefano Maggiolo) wrote…</i></summary>
> What's "it"?
</details>
Fixed.

---

*[cms/io/web_service.py, line 72 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKo8xDaJg6Hc-u5TF1p:-KKpYYCLN3kgnMlPRqtH:-499146009) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/io/web_service.py#L72)):*
<details><summary><i>Previously, stefano-maggiolo (Stefano Maggiolo) wrote…</i></summary>
> Make the comment more useful, explaining why, or what "proper" means?
</details>
Done.

---

*[cms/server/admin/authentication.py, line 74 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKoApqUT81cPK-3zHYF:-KKpYZWoNG-1KvD0FAHk:1758359711) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/server/admin/authentication.py#L74)):*
<details><summary><i>Previously, stefano-maggiolo (Stefano Maggiolo) wrote…</i></summary>
> Typo
</details>
Fixed.

---

*[cms/server/admin/authentication.py, line 117 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKgv6w5-aFfc1qronpG-r1-117:-KKpXvTd29eJlrVmMqZn:1705415099) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/server/admin/authentication.py#L117)):*
<details><summary><i>Previously, codacy-bot (Codacy Bot) wrote…</i></summary>
> ![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [An attribute defined in cms.server.admin.authentication line 64 hides this method](https://www.codacy.com/app/s-maggiolo/cms/file/2792810740/issues/source?bid=3167167&fileBranchId=3370090#l117)
</details>
That's intended.

---

*[cms/server/admin/authentication.py, line 147 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKoBZ1aUnjIHNOgTjMp:-KKpY_tOUfIqt8urrhuv:-499146009) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/server/admin/authentication.py#L147)):*
<details><summary><i>Previously, stefano-maggiolo (Stefano Maggiolo) wrote…</i></summary>
> """Check that the cookie is valid, and if not clear it"""
</details>
Done.

---

*[cms/server/admin/authentication.py, line 150 \[r1\]](https://reviewable.io:443/reviews/cms-dev/cms/623#-KKoBpWxJKbtLlVLqWaf:-KKpYaQUE2Z5mQYsmMbz:1915140605) ([raw file](https://github.com/cms-dev/cms/blob/ccd1137ab3606a119957888327b40e4723209f4b/cms/server/admin/authentication.py#L150)):*
<details><summary><i>Previously, stefano-maggiolo (Stefano Maggiolo) wrote…</i></summary>
> What's this for?
</details>
Explained.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cms-dev/cms/623)*
<!-- Sent from Reviewable.io -->
